__tests__/components/Trunc.spec.js

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import { render } from '@testing-library/react-native';
+import Trunc from '../../src/components/organisms/Trunc';
+
+describe('components/Trunc', () => {
+    const runTruncTest = (content, type, length, expectedResult) => {
+        // Arrange:
+        const props = {
+            type,
+            length,
+        };
+
+        // Act:
+        const screen = render(<Trunc {...props}>{content}</Trunc>);
+        const result = screen.baseElement.children[0].children[0];
+
+        // Assert:
+        expect(result).toBe(expectedResult);
+    };
+
+    describe('children prop is not a string', () => {
+        // Arrange:
+        const type = null;
+        const length = 6;
+
+        test('render empty string', () => {
+            // Arrange:
+            const content = 123;
+
+            // Act + Assert:
+            const expectedResult = '';
+            runTruncTest(content, type, length, expectedResult);
+        });
+    });
+
+    describe('custom type', () => {
+        // Arrange:
+        const type = null;
+        const length = 6;
+
+        test('render full string when its length is equal to length prop', () => {
+            // Arrange:
+            const content = '123456';
+
+            // Act + Assert:
+            const expectedResult = '123456';
+            runTruncTest(content, type, length, expectedResult);
+        });
+
+        test('render truncated string when its length is greater than length prop', () => {
+            // Arrange:
+            const content = '12345678';
+
+            // Act + Assert:
+            const expectedResult = '123456...';
```

Comment:
can we turn the actual `...` (three dots) into ellipsis? https://en.wikipedia.org/wiki/Ellipsis
(`&hellip;`)

Reply:
@gimre-xymcity I tried to replace `...` with `&hellip;` but it is rendered as `&hellip;` text. Not sure if it can work on native.

Reply:
ok, let's leave as is